pkg/io/secrets/secret_locations.go

```diff
@@ -0,0 +1,65 @@
+package secrets
+
+import (
+	"github.com/jenkins-x/jx/pkg/kube"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const vaultSecretsMarker = "useVaultForSecrets"
+
+type SecretLocation interface {
+	// InVault returns whether secrets are stored in Vault
+	InVault() bool
+	// SetInVault sets whether secrets are stored in Vault or not
+	SetInVault(useVault bool)
+}
+
+type secretLocation struct {
+	kubeClient kubernetes.Interface
+	namespace  string
+	usingVault *bool // use a tri-state boolean. nil means uninitialised (so need to lookup from cluster)
+}
+
+// NewSecretLocation creates a SecretLocation
+func NewSecretLocation(kubeClient kubernetes.Interface, namespace string) SecretLocation {
+	return &secretLocation{
+		kubeClient: kubeClient,
+		namespace:  namespace,
+	}
+}
+
+// UsingVaultForSecrets returns true if the cluster has been configured to store secrets in vault
+func (s *secretLocation) InVault() bool {
+	if s.usingVault == nil {
+		configMap := getInstallConfigMap(s.kubeClient, s.namespace)
+		b := configMap[vaultSecretsMarker] != ""
```

Comment:
I think you need to handle nil. It case the configmap does not exists, the `getInstallConfigMap` will only log a message and return nil.

Reply:
good call

Reply:
Great! Thanks